datarobot_batch_scoring/reader.py

```diff
@@ -432,6 +434,25 @@ def sniff_dialect(sample, encoding, sep, skip_dialect, ui):
     return dialect
 
 
+def get_opener_and_mode(is_gz, text=False):
+    if is_gz:
+        if six.PY2:
+            mode = 'r' if text else 'rb'
+            return (gzip_with_encoding, mode)
+        else:
+            mode = 'rt' if text else 'rb'
+            return (gzip.open, mode)
+    else:
```

Comment:
It should work, however, I think the whole `else` can be written using `io`:

```python
mode = 'rt' if text else 'rb'
return (io.open, mode)
```

because:

* `io.open` uses universal newlines by default (no need to pass `U` in openning mode)
* `io.open` is `open` in Python 3